### PR TITLE
py-torch: allow v0.4.1 to build despite missing nervanagpu repo

### PR DESCRIPTION
**Pull request by aweits**
adds the 'submodules_delete' parameter to git versioning so that a
submodule can be removed from the repo.

### CODE REVIEW

File: lib/spack/spack/test/git_fetch.py [hunk 1]
```diff
@@ -217,3 +216,53 @@ def test_get_full_repo(get_full_repo, git_version, mock_git_repository,
...
+    type_of_test = 'tag-branch'
+    t = mock_git_repository.checks[type_of_test]
+
+    # Construct the package under test
+    spec = Spec('git-test')
+    spec.concretize()
+    pkg = spack.repo.get(spec)
+    args = copy.copy(t.args)
+    args['submodules'] = True
+    args['submodules_delete'] = ['third_party/submodule']
```
Comment (adamjstewart) [inline lib/spack/spack/test/git_fetch.py]: Can you add a test for 2+ submodules? Right now you only test 1.

File: lib/spack/docs/packaging_guide.rst [hunk 2]
```diff
@@ -929,6 +929,9 @@ Git fetching supports the following parameters to ``version``:
 * ``tag``: Name of a tag to fetch.
 * ``commit``: SHA hash (or prefix) of a commit to fetch.
 * ``submodules``: Also fetch submodules recursively when checking out this repository.
+* ``submodules_delete``: An array of submodules to forcibly delete from the repository
```
Comment (adamjstewart) [inline lib/spack/docs/packaging_guide.rst]:
```suggestion
* ``submodules_delete``: A list of submodules to forcibly delete from the repository
```
In reality, it's any iterable, but list is the most common iterable in Python.

File: lib/spack/docs/packaging_guide.rst [hunk 3]
```diff
@@ -929,6 +929,9 @@ Git fetching supports the following parameters to ``version``:
 * ``tag``: Name of a tag to fetch.
 * ``commit``: SHA hash (or prefix) of a commit to fetch.
 * ``submodules``: Also fetch submodules recursively when checking out this repository.
+* ``submodules_delete``: An list of submodules to forcibly delete from the repository
```
Comment (adamjstewart) [inline lib/spack/docs/packaging_guide.rst]:
```suggestion
* ``submodules_delete``: A list of submodules to forcibly delete from the repository
```

File: lib/spack/spack/test/conftest.py [hunk 4]
```diff
@@ -725,11 +725,31 @@ def mock_archive(request, tmpdir_factory):
...
 def mock_git_repository(tmpdir_factory):
-    """Creates a very simple git repository with two branches and
-    two commits.
+    """Creates a very simple git repository with two branches,
+    two commits and five submodules.
     """
     git = spack.util.executable.which('git', required=True)
 
+    suburls = []
+    for submodule_count in range(5):
```
Comment (scheibelp) [inline lib/spack/spack/test/conftest.py]: A few minor requests:

* Generate fewer submodules: as many as are necessary to test what you need. I think two would be sufficient.
* Generate different contents in each submodule (can just be different file names)
* Mention in the docstring that each submodule has one commit
* In the docstring "very simple" -> "simple" (there are simpler git repos)

**Comment by scheibelp** [comment]
Generally speaking this looks great. If you don't think https://github.com/spack/spack/pull/14370#discussion_r368191299 would be an improvement let me know.

Once that and https://github.com/spack/spack/pull/14370#discussion_r368165440 are addressed this will be good to go.

File: lib/spack/docs/packaging_guide.rst (hunk 3 above)
Comment (aweits) [reply lib/spack/docs/packaging_guide.rst]: Done

File: lib/spack/spack/test/conftest.py (hunk 4 above)
Comment (aweits) [reply lib/spack/spack/test/conftest.py]: Done

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**